src/auth/auth.controller.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Controller,
+  HttpStatus,
+  Get,
+  Query,
+  HttpException,
+  Post,
+  UseGuards,
+  Req,
+} from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { JwtAuthGuard } from './jwt-auth.guard';
+import { TRequest } from 'src/common/types/user.types';
+
+@ApiTags('Auth')
+@Controller()
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @Get('oauth/callback')
+  async githubAuthCallback(@Query('code') code: string) {
+    try {
+      const jwtToken = await this.authService.handleGitHubCallback(code);
+
+      return jwtToken;
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @ApiBearerAuth()
+  @UseGuards(JwtAuthGuard)
+  @Post('logout')
+  async logout(@Req() req: TRequest) {
+    try {
+      return await this.authService.logout(req);
+    } catch (error) {
+      if (error.status)
+        throw new HttpException(error.message, error.status);
+      else throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+    }
```

Comment:
PrimathonAICodeReviewer

Ensure consistent error handling approach in the `logout` method by always using curly braces for if statements.
```diff
+       // Use curly braces consistently for if statements
```

#### Overall Comments
The code additions look good in terms of functionality. However, improving error handling consistency and adding more detailed error messages can enhance the robustness of the authentication flow. 

Make sure to address the review comments provided above for better maintainability and clarity in error handling. 

Feel free to reach out if you need further assistance or clarification on the suggested changes!


<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->